src/main/java/org/candlepin/subscriptions/tally/TallySnapshotController.java

```diff
@@ -100,6 +102,14 @@ public void produceSnapshotsForAccounts(List<String> accounts) {
                 attemptCloudigradeEnrichment(accounts, accountCalcs);
             }
         }
+        catch (NullPointerException e) {
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            e.printStackTrace(pw);
+            log.error("Could not collect existing usage snapshots for accounts {}, stacktrace: {}",
+                accounts, pw);
+            return;
+        }
```

Comment:
This from Lindsey.  Thanks, Lindsey.

Reply:
I think this style of logging is fine to use in a one-off branch to verify an issue.

However, it's not the idiomatic way to log a stack trace w/ slf4j. We're actually already doing that... we should track down why the *existing* logging statement below:

```java
log.error("Could not collect existing usage snapshots for accounts {}", accounts, e);
```
didn't emit the stack trace (out of scope for this fix - I made https://issues.redhat.com/browse/ENT-3574 for this). For now, please remove these changes.

Reply:
FWIW, the reason the stack trace was not logged could be as simple as https://stackoverflow.com/a/9387598, meaning there would have been at least one earlier instance where the trace was logged.

Reply:
Fair.  I will update this and the new ticket, and remove this block.